sktime/benchmarking/benchmarking_dataclasses.py

```diff
@@ -0,0 +1,430 @@
+    means : list of ScoreResult
+        The mean scores across all folds for each scorer.
+    stds : list of ScoreResult
+        The standard deviation of scores across all folds for
+        each scorer.
+    """
+
+    model_id: str
+    task_id: str
+    folds: dict[int, FoldResults]
+    means: list[ScoreResult] = field(init=False)
+    stds: list[ScoreResult] = field(init=False)
+
+    def __post_init__(self):
+        """Calculate mean and std for each score."""
+        self.means = []
+        self.stds = []
+        scores = {}
+        for fold_idx, fold in self.folds.items():
+            for score in fold.scores:
+                if score.name not in scores:
+                    scores[score.name] = []
+                scores[score.name].append(score.score)
+        for name, score in scores.items():
+            # TODO mean wrongly calculated over all axis.
+            self.means.append(ScoreResult(name, np.mean(score, axis=0)))
+            self.stds.append(ScoreResult(name, np.std(score, axis=0)))
+
+
+@dataclass
+class BenchmarkingResults:
+    """Results of a benchmarking run.
+
+    Parameters
+    ----------
+    results : list of ResultObject
+        The results of the benchmarking run.
+    """
+
+    path: str
+    results: list[ResultObject] = field(default_factory=list)
+
+    def __post_init__(self):
+        """Save the results to a file."""
+        self.storage_backend = get_storage_backend(self.path)
+        self.results = self.storage_backend(self.path).load()
+
+    def save(self):
+        """Save the results to a file."""
+        self.storage_backend(self.path).save(self.results)
+
+    def contains(self, task_id: str, model_id: str):
+        """
+        Check if the results contain a specific task and model.
+
+        Parameters
+        ----------
+        task_id : str
+            The task ID.
+        model_id : str
+            The model ID.
+        """
+        return any(
+            [
+                result.task_id == task_id and result.model_id == model_id
+                for result in self.results
+            ]
+        )
+
+
+def asdict(obj, *, dict_factory=dict, pd_orient="list"):
+    """Return the fields of a dataclass as a dict.
+
+    # Copied from dataclasses.asdict
+
+    """
+    if not hasattr(type(obj), "__dataclass_fields__"):
+        raise TypeError("asdict() should be called on dataclass instances")
+    return _asdict_inner(obj, dict_factory, pd_orient)
+
+
+def _asdict_inner(obj, dict_factory, pd_orient):
+    # Copied from dataclasses._asdict_inner and slightly modified
+    if hasattr(type(obj), "__dataclass_fields__"):
+        result = []
+        for f in fields(obj):
+            value = _asdict_inner(getattr(obj, f.name), dict_factory, pd_orient)
+            result.append((f.name, value))
+        return dict_factory(result)
+    elif isinstance(obj, tuple) and hasattr(obj, "_fields"):
+        # obj is a namedtuple.  Recurse into it, but the returned
```

Comment:
is the problem that you are trying to solve the same that `get_params` in `skbase` solves?

Reply:
The get_params implementation of the base class?